src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -487,11 +487,16 @@ std::pair<std::time_t, std::size_t> restart_info(const RestartIO::RstState * rst
 
     void Schedule::iterateScheduleSection(const ParseContext& parseContext , ErrorGuard& errors, const SCHEDULESection& section , const EclipseGrid& grid,
                                           const FieldPropsManager& fp) {
-        size_t currentStep = 0;
+        size_t currentStep;
         const auto& unit_system = section.unitSystem();
         std::vector<std::pair< const DeckKeyword* , size_t> > rftProperties;
         size_t keywordIdx = 0;
 
+        if (this->m_timeMap.skiprest())
+            currentStep = 0;
+        else
+            currentStep = this->m_timeMap.restart_offset();
+
```

Comment:
This should declare `currentStep` too, to avoid separate declaration and initialisation.  Making `currentStep` a `const` object then depends on how it's used elsewhere.
```C++
size_t currentStep = this->m_timeMap.skiprest()
    ? 0 : this->m_timeMap.restart_offset();
```

Reply:
I'll leave that one as it is - I find the ternary operator quite painful to read. The `currentStep` variable is updated in the routine and can not be `const`

Reply:
> I'll leave that one as it is

Then can you at least move the declaration of `currentStep` down to just above the block where it's initialised?